Update F# Interactive Dependency Manager Plugins README.md (#16261)
dotnet/fsharp: src/FSharp.DependencyManager.Nuget/README.md

<!-- this content is pointed at by url https://aka.ms/dotnetdepmanager please keep the F# team in the loop to update the redirect if the file is moved -->

# F# Interactive Dependency Manager Plugins

Since F# 5.0, `dotnet fsi` and `FsiAnyCPU.exe` (the .NET Framework variant) ships with dependency manager plugins support that can be called like so:

```fsharp
#r "myextension: my extension parameters"
```

For F# Interactive to load the extensions[^1], they either need to:
* be deployed next to `fsi.dll` of dotnet SDK (or next `FsiAnyCPU.exe` for .NET Framework)
* refered to via `--compilertool:<extensionsfolderpath>`[^2] argument

The same goes for the tooling hosting [F# Compiler Service](https://fsharp.github.io/fsharp-compiler-docs/fcs/).

The initial [RFC](https://github.com/fsharp/fslang-design/blob/main/tooling/FST-1027-fsi-references.md) for this feature overviews how it is designed.

> More about F# Interactive References

Reviewer: ```suggestion
**More about F# Interactive References**
```

> The current implementation follows pattern that can be deducted by refering to [implementation in DependencyProvider.fs](https://github.com/dotnet/fsharp/blob/b9687a58cee795a94eb88cf84e309767cc25f6cb/src/Compiler/DependencyManager/DependencyProvider.fs#L145-L322); due to the system working without having a staticically linked dependency, it uses a late binding approach leveraging runtime reflection to identify if an extension conforms to patterns understood by the specifics of the compiler implementation.

Reviewer: ```suggestion
The current implementation follows pattern that can be deducted by referring to [implementation in DependencyProvider.fs](https://github.com/dotnet/fsharp/blob/b9687a58cee795a94eb88cf84e309767cc25f6cb/src/Compiler/DependencyManager/DependencyProvider.fs#L145-L322); due to the system working without having a statically linked dependency, it uses a late binding approach leveraging runtime reflection to identify if an extension conforms to patterns understood by the specifics of the compiler implementation.
```

Author: oh my bad :D 

I'm thinking the nuget one (which is kind of template for now) should provide named SRTP constraints in a separate file that extensions can easily pull (via paket git dependencies or such), to make it statically checked despite it is not linked.

for r in MyCsv.GetSample().Rows do
  printfn "%i = %s" r.X r.Y
```

> [^1]: [Referencing packages in F# Interactive](https://learn.microsoft.com/en-us/dotnet/fsharp/tools/fsharp-interactive/#referencing-packages-in-f-interactive)

Reviewer: ```suggestion
[^1]: [Referencing packages in F# Interactive](https://learn.microsoft.com/dotnet/fsharp/tools/fsharp-interactive/#referencing-packages-in-f-interactive)
```

> [^2]: [F# Interactive options](https://learn.microsoft.com/en-us/dotnet/fsharp/language-reference/fsharp-interactive-options)

Reviewer: ```suggestion
[^2]: [F# Interactive options](https://learn.microsoft.com/dotnet/fsharp/language-reference/fsharp-interactive-options)
```
